app/views/sparql/index.html.erb

```diff
@@ -0,0 +1,90 @@
+      </div>
+      <br>
+      <div>
+        <label>Output Format:</label>
+        <%= form.select :format, [['Table', 'table'], ['JSON', 'json'], ['XML', 'xml']], {}, {style: "margin-left: 10px;"} %>
+      </div>
+      <br>
+      <div>
+        <%= form.submit "Execute Query", style: "background: #007bff; color: white; padding: 8px 16px; border: none;" %>
+        <button type="button" onclick="document.querySelector('textarea').value = ''" style="margin-left: 10px; padding: 8px 16px;">Clear</button>
+        
+        <% if @example_queries && @example_queries.any? %>
+          <select id="example-queries" style="margin-left: 10px; padding: 8px;">
+            <option value="">Load Example Query...</option>
+            <% @example_queries.each do |key, query_data| %>
+              <option value="<%= key %>" data-query="<%= CGI.escapeHTML(query_data['query'].to_s.strip) %>" data-title="<%= CGI.escapeHTML(query_data['title']) %>" data-description="<%= CGI.escapeHTML(query_data['description']) %>">
+                <%= query_data['title'] %>
+              </option>
+            <% end %>
+          </select>
+        <% end %>
+      </div>
+    <% end %>
+<% end %>
+</div>
+
+
+<% if @error %>
+  <div style="background: #ffebee; border: 1px solid #f44336; padding: 15px; margin-top: 20px;">
+    <h4 style="color: #f44336;">Query Error:</h4>
+    <pre style="color: #f44336;"><%= @error %></pre>
+  </div>
+<% end %>
+
+<% if @results && @results.any? %>
+  <div style="background: #e8f5e8; border: 1px solid #4caf50; padding: 15px; margin-top: 20px;">
+    <h4 style="color: #4caf50;">Query Results (<%= @result_count %> results):</h4>
+
+    <% if @format == 'table' && @results.first.is_a?(Hash) %>
+      <div style="overflow-x: auto; max-width: 100%;">
+        <table border="1" style="min-width: 100%; border-collapse: collapse; white-space: nowrap;">
+          <thead>
+          <tr>
+            <% @results.first.keys.each do |key| %>
+              <th style="background: #f5f5f5; padding: 8px; min-width: 150px;"><%= key %></th>
+            <% end %>
+          </tr>
+          </thead>
+          <tbody>
+          <% @results.each do |row| %>
+            <tr>
+              <% row.values.each do |value| %>
+                <td style="padding: 8px; min-width: 150px;">
+                  <% if value.to_s.start_with?('http') %>
+                    <a href="<%= value %>" target="_blank" class="external-link" style="max-width: 300px; overflow: hidden; text-overflow: ellipsis; display: inline-block;"><%= value %></a>
+                  <% else %>
+                    <%= value %>
+                  <% end %>
+                </td>
+              <% end %>
+            </tr>
+          <% end %>
+          </tbody>
+        </table>
+      </div>
+    <% else %>
+      <pre><%= JSON.pretty_generate(@results) %></pre>
+    <% end %>
+  </div>
```

Comment:
this would be cleaner using a partial, or a view helper.

Infact I'm a little confused, as they are exist as partials and are used from `help_documents/index.html` - is that the more recent version but in the wrong place ?